src/main/java/ui/components/FilterTextField.java

```diff
@@ -94,6 +95,20 @@ private void setup() {
     }
 
     /**
+     * Set the background of FilterTextField to white if there is no syntax errors.
+     * Otherwise, background is set to red
+     *
+     * @param isValidFilter
+     */
+    public void setTextFieldBackground(boolean isValidFilter){
+        if (isValidFilter) {
+            this.setStyle("-fx-control-inner-background: white");
+        } else {
+            this.setStyle("-fx-control-inner-background: #EE8993");
```

Comment:
May be declare these two magic strings as constants? If you do that, you don't even need this method. It can be `thisSetStyle(isValidFilter ? VALID_FILTER_STYLE : INVALID_FILTER_STYLE)`